Kuadrant/deployment
# Add observability pieces to hub & workers

Moved from https://github.com/roivaz/kuadrant-deployment/pull/1

* Adds an applicationset for installing prometheus, various monitors and metrics config in each worker cluster
* Adds an application for installing grafana and thanos in the hub cluster
* Inits and links skupper sites for each cluster as part of local-setup target
* Exposes thanos-receive-router service over the skupper link
* configures remoteWrite from workers to the thanos-receive-router svc over skupper
* configures cluster_id label on remoteWrites

To verify:

```bash
make local-setup
```

The new argocd apps for observability hub & worker (x2) should be green.
You should be able to access grafana by port-forward to the svc (user: admin pass: admin).

```bash
kubectl --kubeconfig ./kubeconfig --context kind-kuadrant-local-0 -n monitoring port-forward svc/grafana 3000:3000
```

There's a data source called Kuadrant-Thanos-Hub. 
In the explore view of Grafana, you can verify this is working by running this query:

```
count(ALERTS) by(cluster_id)
```

It should give 2 graph lines, with 2 cluster ids: `in-cluster` & `kuadrant-local-1`
